node/src/components/contract_runtime/types.rs

```diff
@@ -81,6 +81,15 @@ impl From<EraValidatorsRequest> for GetEraValidatorsRequest {
     }
 }
 
+/// Effects from running step and the next era validators that are gathered when an era ends.
+#[derive(Debug)]
+pub struct StepEffectAndUpcomingEraValidators {
+    /// An [`ExecutionEffect`] created by an era ending.
+    pub step_execution_effect: ExecutionEffect,
+    /// Upcoming era validators.
```

Comment:
We could explain in a bit more detail here which validators exactly are included, e.g.:

```suggestion
    /// Validator sets for all upcoming eras that have already been determined.
```